corehq/apps/api/resources/v0_5.py

```diff
@@ -962,13 +966,6 @@ def dispatch(self, request_type, request, **kwargs):
             response = super(BaseODataResource, self).dispatch(
                 request_type, request, **kwargs
             )
-
-            # order REALLY matters for the following code. It should be called
-            # AFTER the super's dispatch or request.couch_user will not be present
```

Comment:
This seems like an important comment, but I don't know enough about location permissions to understand why it's OK to remove it now.

Reply:
I originally added this, and it was more of a hack rather than a proper implementation. I think I didn't understand the auth meta methods for the APIs. This check now happens in the correct order with the changes made to `ODataAuthentication`